README.md

```diff
@@ -42,14 +59,68 @@ var output = postcss([
 console.log(output);
 ```
 
-[*For more general PostCSS usage, see this section*](https://github.com/postcss/postcss#usage)
 
+# Syntax
 
-## At-rule support `@media`, `@support`, etc
+### Defining Custom Properties with `--*`
 
-You can add rules that declare CSS variables nested inside at-rules. You can even nest the at-rules however deep you want.
+A custom property is any property whose name starts with two dashes (U+002D HYPHEN-MINUS). A property must be in a rule.
 
-The following CSS:
+*Note: `:root` is nothing more than the selector for the root DOM node.*
+
+```css
+:root {
+	--foo-width: 100px;
+	--foo-bg-color: rgba(255, 0, 0, 0.85);
+}
+```
+
+You may also declare custom properties in a normal selector. Any other selector like `.class`, `#id`, or even `#foo ~ .bar > span.baz` works.
+
+```css
+.foo {
+	--foo-width: 100px;
+	--foo-bg-color: rgba(255, 0, 0, 0.85);
+}
+```
+
+A custom property can be declared multiple times, but only the last one takes precedence.
+
+```css
+:root {
+	--foo-width: 100px;
+	--foo-width: 200px; /* This is the declaration used. */
+}
+
+.foo {
+	--foo-width: 100px;
+}
+
+.foo {
+	--foo-width: 200px; /* This is the declaration used. */
+}
+```
```

Comment:
```css
:root: {
	--some-color: red;
}

.foo {
	/* red */
	color: var(--some-color);
}


.bar {
	--some-color: blue;
	/* blue */
	color: var(--some-color);
}

.bar:hover {
	--some-color: green;
	/* Automically gets a `color: green;` declaration because we `--some-color` used within scope elsewhere */
}
```